server/grpc/gogoreflection/fix_registration.go

```diff
@@ -18,16 +20,21 @@ func getFileDescriptor(filePath string) []byte {
 		return fd
 	}
 
-	return proto.FileDescriptor(filePath) //nolint:staticcheck
+	return proto.FileDescriptor(filePath) //nolint:staticcheck // keep for backward compatibility
 }
 
-func getMessageType(name string) reflect.Type {
+func getMessageType(name string) (reflect.Type, error) {
 	typ := gogoproto.MessageType(name)
 	if typ != nil {
-		return typ
+		return typ, nil
 	}
 
-	return proto.MessageType(name) //nolint:staticcheck
+	nametype, err := protoregistry.GlobalTypes.FindMessageByName(protoreflect.FullName(name))
+	if err != nil {
+		return nil, err
+	}
+
+	return reflect.TypeOf(proto.MessageV1(nametype.Zero().Interface())), nil
```

Comment:
@testinginprod can i get you to check this

Reply:
this doesn't look safe to do, considering our multiple registries